lib/Models/Catalog/Gtfs/GtfsCatalogItem.ts

```diff
@@ -163,8 +167,31 @@ export default class GtfsCatalogItem extends MappableMixin(
         data.sourceId
       );
 
-      if (!entity.model && this._model) {
-        entity.model = this._model;
+      if (!entity.model) {
+        if (this._coloredModels) {
+          const gtfsEntity: FeedEntity = data.featureInfo?.get("entity");
+          const value = jsonPathLike(
+            gtfsEntity as JsonObject,
+            this.model.colorModelsByProperty.property!
+          );
+          if (value !== undefined) {
+            let breakLoop = false;
+            this.model.colorModelsByProperty.colorGroups.forEach(
+              (colorGroup, i) => {
+                if (breakLoop || colorGroup.regExp === undefined) return;
```

Comment:
Possible bug: `breakLoop` here might not serve its intended purpose of early returning once we hit a matching regexp. The `return` only returns from the arrow function but `forEach` will continue iterating the full array. We might have to rewrite using `for` or `find()`.

Reply:
Yeah good point. I probably want `findIndex`.

Reply:
Fixed